simplepso/pso.py

```diff
@@ -282,6 +256,9 @@ def run(self, num_particles, num_iterations, num_processors=1,
         else:
             self._setup_pso()
 
+        if not callable(cost_function):
+            raise Exception("Provide a callable cost function")
```

Comment:
you may want to make the exception more specific. Maybe `TypeError`?

Reply:
Done!